### PR TITLE
Update readme with Usage instructions

### LINKED ISSUE #37: Update README file
### Proposed change

- [x] Refine description of the JupyterHub Pytest Plugin project
- [x] Add a Usage section for plugin installation and fixture importation
- [x] Provide an example of plugin integration using the [DockerSpawner draft PR](https://github.com/jupyterhub/dockerspawner/pull/476) to demonstrate usage
- [x] Add workflow status badges

### Who would use this feature?

This feature is useful for users of and contributors to the JupyterHub Pytest Plugin

### PR DESCRIPTION
**Pull request by Sheila-nk**
- refined the project's description
- added a usage section with installation and import instructions
- added an integration example using the `DockerSpawner` [draft PR](https://github.com/jupyterhub/dockerspawner/pull/476)

Closes #37

### CODE REVIEW

**Comment by Sheila-nk** [comment]
Ping @GeorgianaElena 

The ReadTheDocs build is failing and I think it is because of the `pydata-sphinx-theme`  v0.13.0 which was released a couple of days ago. There is this note on the [pypi page](https://pypi.org/project/pydata-sphinx-theme/0.13.0/) that it might not be compatible with the most recent version of `Sphinx` which I believe we are using according to the build logs.

> Note This theme may not work with the latest major versions of Sphinx, especially if they have only recently been released. Please give us a few months of time to work out any bugs and changes when new releases are made.

Do you think we should pin the `pydata-sphinx-theme` extension to the lower version before the compatibility issue is sorted?

**Comment by manics** [comment]
Since RTD is a "production" deployment of our docs we could pin all deps and let dependabot bump them?

If you want we could also switch to `jupyterhub-sphinx-theme` https://github.com/jupyterhub/team-compass/issues/524

**Comment by Sheila-nk** [comment]
@manics Thanks for the suggestion. I have looked over the `jupyterhub-sphinx-theme` and the `JupyterHub Python Repo Template` [Draft PR](https://github.com/jupyterhub/jupyterhub-python-repo-template/pull/8) using it. I like it and I think we should go ahead and make the switch too. 
* make a draft PR using the `jupyterhub-sphinx-theme`
* create a `dependabot.yml` configuration file to update dependencies

File: README.md
Comment (manics) [inline README.md:11]:
```suggestion
[![GitHub Workflow Status - Test](https://github.com/jupyterhub/pytest-jupyterhub/actions/workflows/test.yml/badge.svg)](https://github.com/jupyterhub/pytest-jupyterhub/actions)
```
https://img.shields.io/github/workflow/status/jupyterhub/pytest-jupyterhub/Test?logo=github&label=tests doesn't exist

File: README.md
Comment (manics) [inline README.md:32]: Is it worth deleting this "Example"? It's a bit confusing having two"Example" sections in the README, and this one goes into a lot of technical detail. The usage example seems much more relevant and helpful to readers.

File: README.md
Comment (Sheila-nk) [reply README.md:12]: I am not sure what you mean by 'public' but we have not set up the project on codecov yet. I don't think I have the relevant permissions for that. I can delete this badge for now until that is done.

Comment (Sheila-nk) [reply README.md:12]: I am not sure what you mean by 'public' but we have not set up the project on codecov yet. I don't think I have the relevant permissions for that. I can delete this badge for now until that is done.

File: README.md
Comment (Sheila-nk) [reply README.md:32]: Okay, I agree. I will remove this example section.

File: README.md
Comment (GeorgianaElena) [reply README.md]: `codecov` doesn't look to be setup for this repo yet. I believe it makes sense to not have a badge for it.
Also, @Sheila-nk, can you please open up an issue about enabling codecov for this repo?

File: README.md
Comment (GeorgianaElena) [inline README.md:34]: I believe it's worth leaving a note here about the fact that we're instructing to install the package from GitHub because the package was not yet released on PyPI. And then change the install instruction once that's done. Also, adding this to a todo list, somewhere in an issue to not forget, will be also a good idea.

File: README.md
Comment (GeorgianaElena) [inline README.md:44]: This is subject to how each project is managing their dependencies, and it's hard and not recommended to cover all cases, so let's remove and leave it to each project to define.

File: README.md
Comment (GeorgianaElena) [inline README.md:51]: When installing a package with `pip install git+https://github.com/jupyterhub/pytest-jupyterhub.git@main`, its dependencies get installed also.

Plus, we're defining the `pytest-jupyterhub`'s deps [in `pyproject.toml`](https://github.com/jupyterhub/pytest-jupyterhub/blob/43a862d0f19e07889c96ff7c48b26effcfff325d/pyproject.toml#L28-L29) and don't actually have a `requirements.txt` file in this repo, right?

File: README.md
Comment (GeorgianaElena) [inline README.md:59]: The pytest docs describe differently how to use a plugin in a test suite: https://docs.pytest.org/en/7.1.x/how-to/writing_plugins.html#requiring-loading-plugins-in-a-test-module-or-conftest-file

Is there a reason why we're taking a different approach here and importing the fixtures same like we'd do from a regular Python package?

File: README.md
Comment (Sheila-nk) [reply README.md:51]: Oh yes! That is my mistake. All dependencies are defined in `pyproject.toml`. No `requirements.txt` file for this project, except for the docs.

File: README.md
Comment (Sheila-nk) [reply README.md:59]: Hmm, true. I might need to change the `DockerSpawner` [draft PR](https://github.com/jupyterhub/dockerspawner/pull/476) as well since I imported it instead of loading the plugin. Thank you for catching this.

**Comment by Sheila-nk** [comment]
@GeorgianaElena I have worked on the recommended changes and added a `contributing` section to this README. Please review the changes and let me know if they are ready for merging.

File: README.md
Comment (consideRatio) [inline README.md:11]: I want us to consistently use the img.shields.io for consistent badges styling, the img.shields.io badges for github workflows needs an update in many of our github repos to work after a change though. Here is an updated version!

```suggestion
[![GitHub Workflow Status - Test](https://img.shields.io/github/actions/workflow/status/jupyterhub/pytest-jupyterhub/test.yml?logo=github&label=tests&branch=main)](https://github.com/jupyterhub/pytest-jupyterhub/actions)
```

File: README.md
Comment (consideRatio) [inline README.md:38]: This will start working as soon as a git tag has been pushed with #48 merged and credentials to push to a PyPI project are setup.

```suggestion
pip install pytest-jupyterhub
```

**Review by consideRatio — APPROVED** [review]
This looks great to me! I made two code suggestions that I think makes sense.

**Comment by consideRatio** [comment]
@Sheila-nk btw amazing README.md file, I love it! It does a perfect job introducing a reader I think - it doesn't assume too much knowledge ahead of time and onboards the reader, and it doesn't try to explain too much either. Perfectly balanced in my mind!

**Comment by Sheila-nk** [comment]
> @Sheila-nk btw amazing README.md file, I love it! It does a perfect job introducing a reader I think - it doesn't assume too much knowledge ahead of time and onboards the reader, and it doesn't try to explain too much either. Perfectly balanced in my mind!

Thank you @consideRatio for your kind words and review ❤️

**Pull request merged**
